### DISCUSSION TITLE
Select area with mousedown-mouseup

### DISCUSSION THREAD

**Discussion by OoDeLally** [gTile/gTile#387] [Ideas]
Thanks for this amazing app!
Currently selecting an area takes two mouse clicks (1st to select the starting tile, and 2nd to select the expansion of the rectangle).
Would it be possible to bind the starting tile to `mousedown` and validate the area with `mouseup`?
So that it takes only one click to select an area.
Thanks!

**Comment by scherepanov** [comment]
Yes, that would be a good idea. That would be more in-line with mainstream GUI behaviour.
Problem is that historically someone many years ago implemented it this way. Now changing it would be forcing this change to a user base... So it never got changed.
And, original mouse-down + mouse+down interface is not that bad, as it also tracks mouse pointer and reflect it on a screen.

**Comment by cwh1te** [comment]
@scherepanov sorry to necromance this issue, but wouldn't it be possible to implement this behavior only for mouse-down, drag (more than one square), and mouse-up? So if you mouse down and then up the normal behavior applies, but if you move the mouse some distance between then this new behavior would be desirable.
I haven't looked at the code in a couple years so I'm probably forgetting something that prevents this from working...

**Comment by OoDeLally** [comment]
We could even provide it as a optional behavior (a checkbox in the preference panel).
So we dont force this change to the user base.

**Comment by scherepanov** [comment]
Looking at code, that should be pretty straightforward.
Add mouse up event handling. If it happens in current tail, then ignore it, have old behaviour.
If it happens in another tile, treat it as second mouse down.
Looks like it will be 100% compatible with current behaviour.
Any takers? Who want to submit PR?

**Comment by OoDeLally** [comment]
This seems to only be a partial solution. If we do this, assigning a window to one (1) tile will still require two clicks. Right?
I would rather have a checkbox in the settings to actually switch between the two behaviors.

**Comment by scherepanov** [comment]
OK, let's save one click.
Though, I think it will be confusing for casual users.

Personally I am trying to avoid mouse usage as much as possible. I am using .move/resize keyboard shortcuts. First, I am activating gTile with <Super>-[Return]/[[KP-Enter], then using arrows/shift-arrows to move/resize window, then hit enter to place window.
